src/dune_rpc_server/dune_rpc_server.ml

```diff
@@ -202,6 +229,25 @@ module Session = struct
     | Ok { Versioned.Staged.encode } ->
       send t (Some [ Notification (encode n) ])
 
+  let request t decl id req =
+    let* () = Fiber.return () in
+    match V.Handler.prepare_request t.handler decl with
+    | Error error ->
+      Code_error.raise "client doesn't support request"
+        [ ("id", Dune_rpc_private.Id.to_dyn id)
+        ; ("error", Dune_rpc_private.Version_error.to_dyn error)
+        ]
+    | Ok { Versioned.Staged.encode_req; decode_resp } -> (
+      let req = encode_req req in
+      let* ivar = Stage1.request t.base (id, req) in
+      let+ resp = Fiber.Ivar.read ivar in
+      match resp with
+      | Error _ -> Code_error.raise "unexpected error" []
```

Comment:
Why drop the error on the floor? Same below.

Reply:
This error can only occur if we're making a call from dune to an action runner when they're of a different version. So basically, it's possible at the moment. I'll improve the message anyway.